src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -0,0 +1,289 @@
+import net.minecraft.block.properties.PropertyDirection;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Helper class to place and remove constructionTapes from the buildings.
+ */
+public final class ConstructionTapeHelper
+{
+    public static final  PropertyDirection FACING     = BlockHorizontal.FACING;
+
+
+    public static int checkIfPlaceable(@NotNull int x, @NotNull int y, @NotNull int z, @NotNull World world)
+    {
+        int newY = y;
+        boolean working = true;
+        while (working)
+
+        {
+            BlockPos block = new BlockPos(x, newY, z);
+            BlockPos blockMin1 = new BlockPos(x, newY-1, z);
+            if (world.getBlockState(block).isFullBlock() || world.getBlockState(block).getMaterial().isLiquid())
+                newY = newY+1;
+            else
+            if (world.getBlockState(blockMin1).isFullBlock() || world.getBlockState(blockMin1).getMaterial().isLiquid())
+                working = false;
+            else
+                newY = newY-1;
+        }
+        return newY;
+    }
+
+
+    /**
+     * Private Constructor to hide implicit one.
+     */
+    private ConstructionTapeHelper()
+    {
+        /**
+         * Intentionally left empty.
+         */
+    }
+
+    /**
+     * Proxy to place the tape also with the building only.
+     * @param building the building.
+     * @param world the world.
+     */
+    public static void placeConstructionTape(@NotNull AbstractBuilding building, @NotNull World world)
+    {
+        placeConstructionTape(new WorkOrderBuild(building, 1), world);
+    }
+
+
+    public static void placeConstructionTape(@NotNull WorkOrderBuild workOrder, @NotNull World world)
+    {
+        final StructureWrapper wrapper = new StructureWrapper(world, workOrder.getStructureName());
+        final BlockPos pos = workOrder.getBuildingLocation();
+        int tempRotation = 0;
+
+        if (workOrder.getRotation() == 0 && !(workOrder instanceof WorkOrderBuildDecoration))
+        {
+            final IBlockState blockState = world.getBlockState(pos);
+            if (blockState.getBlock() instanceof AbstractBlockHut)
+            {
+                tempRotation = BlockUtils.getRotationFromFacing(blockState.getValue(AbstractBlockHut.FACING));
+            }
+        }
+        else
+        {
+            tempRotation = workOrder.getRotation();
+        }
+
+        wrapper.rotate(tempRotation, world, workOrder.getBuildingLocation());
+        wrapper.setPosition(pos);
+        int x1 = wrapper.getPosition().getX() - wrapper.getOffset().getX() - 1;
+        int z1 = wrapper.getPosition().getZ() - wrapper.getOffset().getZ() - 1;
+        int x3 = wrapper.getPosition().getX() + (wrapper.getWidth() - wrapper.getOffset().getX());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'x3' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)